### PR TITLE
[JENKINS-35976] Workspace cleanup

### PR DESCRIPTION
**Pull request by alexsomai**
Related to issue [[JENKINS-35976]](https://issues.jenkins-ci.org/browse/JENKINS-35976)

Summary of this pull request:
 - [x] Write documentation about workspace cleanup using the ws-cleanup-plugin
 - [x] Write tests for workspace cleanup using the ws-cleanup-plugin

CC @oleg-nenashev @martinda

### CODE REVIEW

**Comment by alexsomai** [comment]
This build will fail until https://github.com/jenkinsci/ws-cleanup-plugin/pull/27 is merged and released.

**Comment by oleg-nenashev** [comment]
It depends on https://github.com/jenkinsci/ws-cleanup-plugin/pull/27, I suppose

File: pom.xml [hunk 1]
```diff
@@ -64,6 +64,12 @@
             <artifactId>workflow-step-api</artifactId>
             <version>${workflow.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>ws-cleanup</artifactId>
+            <version>0.30-SNAPSHOT</version>
+            <optional>true</optional>
```
Comment (oleg-nenashev) [inline pom.xml]: IIRC the decision was to use the dependency for test only. So maybe it should define the "test" <code>scope</code>

Comment (alexsomai) [inline pom.xml]: I remember that we have discussed to make it `optional`, but maybe I've misunderstood.

Comment (oleg-nenashev) [inline pom.xml]: CC @martinda

**Comment by martinda** [comment]
Define the dependency as a test scope.

File: src/test/java/org/jenkinsci/plugins/ewm/TestUtil.java [hunk 2]
```diff
@@ -47,4 +62,28 @@ public int compare(Disk disk1, Disk disk2) {
             }
         });
     }
+
+    public static void addExternalWorkspaceNodeProperty(Node node, String diskPoolRefId, DiskNode... diskNodes) {
```
Comment (oleg-nenashev) [inline src/test/java/org/jenkinsci/plugins/ewm/TestUtil.java]: If it's public, it deserves some Javadoc.
Or maybe the entire class could be made Restricted

Comment (alexsomai) [inline src/test/java/org/jenkinsci/plugins/ewm/TestUtil.java]: This method is used only for testing purposes. 
Is there any need to restrict it in this case?

Comment (alexsomai) [inline src/test/java/org/jenkinsci/plugins/ewm/TestUtil.java]: I've restricted the entire class for external use.

**Comment by alexsomai** [comment]
I think that this is ready for merge?

**Comment by oleg-nenashev** [comment]
👍

**Pull request merged**